services/src/main/java/org/keycloak/services/clientpolicy/condition/UpdatingClientSourceConditionFactory.java

```diff
@@ -74,7 +74,7 @@ public String getId() {
 
     @Override
     public String getHelpText() {
-        return null;
+        return "It uses the source of updating client to determine whether the policy is applied.";
```

Comment:
I suggest to make the help text a bit more clear as it will be displayed in the UI and will be nice to make it a bit more obvious what exactly it means?

For example "The condition checks the context how is client created/updated to determine whether the policy is applied. For example it checks if client is created with admin REST API or OIDC dynamic client registration. And for the letter case if it is ANONYMOUS client registration or AUTHENTICATED client registration with Initial access token or Registration access token and so on."

Also my vote is to rename the class to "UpdatingClientContextCondition" (and similarly rename the provider ID) as the "source" implies that condition is interested about concrete user, who creates/updatest the client rather about the "context" how is the client created/updated (admin REST API, OIDC dynamic client registration etc).

@stianst @valb3r What is your though on the above?

Reply:
@mposolda 
> I suggest to make the help text a bit more clear as it will be displayed in the UI and will be nice to make it a bit more obvious what exactly it means?

Yes, it is a good idea 

> Also my vote is to rename the class to "UpdatingClientContextCondition" (and similarly rename the provider ID) as the "source" implies that condition is interested about concrete user, who creates/updatest the client rather about the "context" how is the client created/updated (admin REST API, OIDC dynamic client registration etc).

Makes sense to me, also `ClientUpdateContextCondition` is clearer as is composed from ClientUpdateContext + Condition that are key entities now for the class

Reply:
@mposolda I agree with your suggesting help text. I'll fix the existing help text.

Reply:
+1 for ClientUpdateContextCondition

Reply:
@valb3r I'd like to rename the class name and its provider id as you proposed. @mposolda  WDYT?

Reply:
@mposolda @valb3r Thank you. I'll revise codes as you proposed.